packages/flutter/lib/src/material/material_state.dart

```diff
@@ -407,6 +407,97 @@ abstract class MaterialStateOutlinedBorder extends OutlinedBorder implements Mat
   /// widget or theme.
   @override
   OutlinedBorder? resolve(Set<MaterialState> states);
+
+  /// Creates a [MaterialStateOutlinedBorder] from a
+  /// [MaterialPropertyResolver<OutlinedBorder?>] callback function.
+  ///
+  /// If used as a regular [OutlinedBorder], the border resolved in the default state
+  /// (the empty set of states) will be used.
+  ///
+  /// Usage:
+  ///
+  /// ```dart
+  /// ChipTheme(
+  ///   data: Theme.of(context).chipTheme.copyWith(
+  ///     shape: MaterialStateOutlinedBorder.resolveWith((Set<MaterialState> states) {
+  ///       if (states.contains(MaterialState.selected)) {
+  ///         return const StadiumBorder(
+  ///           side: BorderSide(color: Colors.transparent),
```

Comment:
This seems like a unusual use case. Presumably what you want is a 1 pixel margin around the stadium border. That's OK however it's a little subtle. Maybe add a comment explaining what the goal is or demo a more straightforward use case. Here and in the other example.

Reply:
this may not be a good example indeed, I'll think about a better one.

Reply:
I checked the `resolveWith` from other `WidgetStateX` classes and none provide examples, so I've removed it from here as well.